### DISCUSSION TITLE
Reducing @next directory size in docker image

### DISCUSSION THREAD

**Discussion by ripvannwinkler** [vercel/next.js#29653] [Help]
A recently developed app with Next.js 11.1.2, when build with docker results in a docker image with the following directory sizes:

	/app # du -a /app | sort -n -r | head -n 20
	365608  /app
	335244  /app/node_modules
	128236  /app/node_modules/@next
	62000   /app/node_modules/typescript
	58816   /app/node_modules/typescript/lib
	44236   /app/node_modules/next
	44072   /app/node_modules/next/dist
	36216   /app/node_modules/next/dist/compiled
	32824   /app/node_modules/@next/swc-win32-x64-msvc
	32812   /app/node_modules/@next/swc-win32-x64-msvc/next-swc.win32-x64-msvc.node
	32372   /app/node_modules/@next/swc-darwin-x64
	32360   /app/node_modules/@next/swc-darwin-x64/next-swc.darwin-x64.node
	31872   /app/node_modules/@next/swc-linux-x64-gnu
	31860   /app/node_modules/@next/swc-linux-x64-gnu/next-swc.linux-x64-gnu.node
	29936   /app/node_modules/@next/swc-darwin-arm64
	29924   /app/node_modules/@next/swc-darwin-arm64/next-swc.darwin-arm64.node

Clearly, SWC is causing my docker image to balloon in size. Is there a way to reduce this? Can I specify the target OS and exclude the other SWC bindings? Please help.

**Comment by timneutkens** [comment]
Only the supported version of the swc binding should be installed, what package manager are you using?

**Reply by ripvannwinkler** [reply]
Plain old NPM.  My docker file literally just does `npm ci --no-progress` and then `npm run build`. The directory is empty to start, and the node_modules folder from the dev source is included in both .gitignore and .dockerignore.
